core/state_connector_test.go

```diff
@@ -0,0 +1,153 @@
+			}
+			if m.testCase == "happy_path" {
+				return nil, 0, nil
+			}
+			return nil, 0, errors.New("finalization error")
+		}
+	case m.testCase == "default_attestors_error_1":
+		return nil, 0, errors.New("voter whitelister error")
+	case m.testCase == "default_attestors_error_2":
+		switch addr {
+		case common.HexToAddress(GetPrioritisedFTSOContract(nil)):
+			return []byte("voter"), 0, nil
+		case common.BytesToAddress([]byte("voter")):
+			return nil, 0, errors.New("list price provider error")
+		}
+	}
+	return nil, 0, errors.New("undefined test case")
+}
+
+func (m *mockStateConnectorCaller) WithBlockContext(bc vm.BlockContext) { m.context = bc }
+
+func (m *mockStateConnectorCaller) BlockContext() vm.BlockContext { return m.context }
+
+func TestStateTransition_FinalisePreviousRound(t *testing.T) {
+	t.Run("happy_path", func(t *testing.T) {
+		mockMsg := &mockMessage{
+			from: common.BytesToAddress([]byte("happy_path")),
+		}
+		mockSCC := &mockStateConnectorCaller{
+			testCase: "happy_path",
+			msg:      mockMsg,
+			context: vm.BlockContext{
+				Coinbase: common.BigToAddress(big.NewInt(1000)),
+			},
+		}
+		c := newConnector(mockSCC, mockMsg)
+		currentRoundNumber := []byte("222")
+		err := c.finalisePreviousRound(big.NewInt(10), big.NewInt(10), currentRoundNumber)
+		assert.NoError(t, err)
+		assert.Equal(t, common.BigToAddress(big.NewInt(1000)), mockSCC.context.Coinbase, "coinbase address should be changed to the original address")
+	})
+	t.Run("default_attestors_error_1", func(t *testing.T) {
+		mockMsg := &mockMessage{
+			from: common.BytesToAddress([]byte("default_attestors_error_1")),
+		}
+		mockSCC := &mockStateConnectorCaller{
+			testCase: "default_attestors_error_1",
+			msg:      mockMsg,
+		}
+		c := newConnector(mockSCC, mockMsg)
+		currentRoundNumber := []byte("222")
+		err := c.finalisePreviousRound(big.NewInt(10), big.NewInt(10), currentRoundNumber)
+		assert.EqualError(t, err, "failed to get VoterWhitelister contract: voter whitelister error")
+	})
+	t.Run("default_attestors_error_2", func(t *testing.T) {
+		mockMsg := &mockMessage{
+			from: common.BytesToAddress([]byte("default_attestors_error_2")),
+		}
+		mockSCC := &mockStateConnectorCaller{
+			testCase: "default_attestors_error_2",
+			msg:      mockMsg,
+		}
+		c := newConnector(mockSCC, mockMsg)
+		currentRoundNumber := []byte("222")
+		err := c.finalisePreviousRound(big.NewInt(10), big.NewInt(10), currentRoundNumber)
+		assert.EqualError(t, err, "failed to get FTSO price providers: list price provider error")
+	})
+	t.Run("finalized_data_error", func(t *testing.T) {
+		mockMsg := &mockMessage{
+			from: common.BytesToAddress([]byte("finalized_data_error")),
+		}
+		mockSCC := &mockStateConnectorCaller{
+			testCase: "finalized_data_error",
+			msg:      mockMsg,
+			context: vm.BlockContext{
+				Coinbase: common.BigToAddress(big.NewInt(1000)),
+			},
+		}
```

Comment:
These could be created using a helper function, with just 2 arguments:

```suggestion
		msg, scc := buildStateConnectorMock("finalized_data_error", 1000)
		scc.CallFunc = func() { // Override call function for specific test case.
```

And buildStateConnectorMock would be as simple as

```go
func buildStateConnectorMock(from string, address int) (*mockMessage, *mockStateConnectorCaller) {
	msg := mockMessage{
		from: common.BytesToAddress([]byte(from)),
	}
	scc := mockStateConnectorCaller{
		testCase: "finalized_data_error",
		msg:      mockMsg,
		context: vm.BlockContext{
			Coinbase: common.BigToAddress(big.NewInt(address)),
		},
	}
	return &msg, &scc
}
```